core/src/main/java/hudson/model/WorkspaceCleanupThread.java

```diff
@@ -139,6 +139,14 @@ private boolean shouldBeDeleted(@Nonnull TopLevelItem item, FilePath dir, @Nonnu
             }
         }
 
+        // TODO this may only check the last build in fact:
+        if (item instanceof Job<?,?>) {
+            Job<?,?> j = (Job<?,?>) item;
+            if (j.isBuilding()) {
+                return false;
```

Comment:
Fine logging message here would be useful

Reply:
(Added fine logging message in 4th commit)